homeassistant/components/plugwise/binary_sensor.py

```diff
@@ -35,19 +46,22 @@ class PlugwiseBinarySensorEntityDescription(BinarySensorEntityDescription):
         icon="mdi:hvac",
         icon_off="mdi:hvac-off",
         entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data["binary_sensors"]["compressor_state"],
```

Comment:
As per Frencks (in #93155) and you suggestion I rewrote the approach, `models.py` no already shows the intended parts for `sensor` and `switch` as well. I know we should stick to only `binary_sensor` as per keeping it small, but we would like to know if this is going in the right direction.

I took you suggestion of passing the key and combined it to leave lambda's out as requested in the other draft.

Reply:
I really think the code became worse after this part. Like the suggestions made here by epenet have been taken and over-engineered IMHO.

The whole lookup stuff is really nog needed at all. All epenet suggested, was simply passing part of the dictionary, instead of all.